src/http/server.ts

```diff
@@ -437,44 +250,63 @@ View more information regarding this deprecation/removal at:
    * @returns The response as stringified JSON. This is only used for
    * unit testing purposes.
    */
-  public handleHttpRequestForFavicon(
+  public async handleHttpRequestForFavicon(
     request: Drash.Http.Request,
-  ): Drash.Interfaces.ResponseOutput {
-    const headers = new Headers();
-    headers.set("Content-Type", "image/x-icon");
-
-    const response = new Drash.Http.Response(request);
+  ): Promise<Drash.Interfaces.ResponseOutput> {
+    const response = this.buildResponse(request);
+    response.body = "";
+    response.headers = new Headers();
     response.status_code = 200;
-    response.headers = headers;
+
+    response.headers.set("Content-Type", "image/x-icon");
 
     try {
-      const body = Deno.readFileSync(`${Deno.realPathSync(".")}/favicon.ico`);
-      response.body = body;
-      if (!this.trackers.requested_favicon) {
-        this.trackers.requested_favicon = true;
-        this.logDebug("/favicon.ico requested.");
-        if (!body) {
-          this.logDebug("/favicon.ico was not found.");
-        } else {
-          this.logDebug("/favicon.ico was found.");
-        }
-        this.logDebug(
-          "All future log messages for /favicon.ico will be muted.",
-        );
-      }
+      response.body = Deno.readFileSync(
+        `${Deno.realPathSync(".")}/favicon.ico`,
+      );
     } catch (error) {
-      response.body = "";
     }
 
-    let output: Drash.Interfaces.ResponseOutput = {
-      body: response.body as string,
-      headers: headers,
-      status: response.status_code,
-    };
+    return response.send();
+  }
```

Comment:
There is no asynchronous code in this function... Now, instead of making it a normal function, I would keep the `async` signature and add asynchronous code (it's always better). How? Make the file access stuff using their asynchronous methods.
```ts
const path = `${await Deno.realPath(".")}/favicon.ico`;
response.body = await Deno.readFile(path)
```

Reply:
Actually, this happens in a bunch of places. Here's a list:

- handleHttpRequestForStaticPathAsset
- handleHttpRequestForVirtualPathAsset

Reply:
6a35af7